### ISSUE TITLE
feat(CNLS/CNLSDDF): Implementation of get_frontier.

### ISSUE THREAD

**Issue by JulianATA** [ds2010/StoNED-Python#16]
The `get_frontier` function is for getting the value of estimated frontier(y value) by CNLS/CNLSDDF.
Here is the some thought for better implementation of `get_frontier`.
Please help me justify if my thought have some logical error.

Since true y value = estimated y value + residual for additive models, we may implement the frontier like below:
## CNLS 
The fallowing y refer to the `true y value`; frontier refer to `estimated y value`.
### Additive
```
frontier = y - residual
```
### Multiplicative
```
frontier = y/(exp(residual)) -1
```
## CNLSDDF
The fallowing y refer to the `true y value`; frontier refer to `estimated y value`.

```
frontier list = y list - residual list
```

**Comment by ds2010** [comment]
For the multiplicative case, I think the estimated frontier should be `model.frontier[i] + 1` thanks to the variable `self.__model__.frontier` that we have defined in the class CNLS. The `get_frontier`:

       def get_frontier(self):
        """Return estimated frontier value by array"""
        if self.optimization_status == 0:
            self.optimize()
        frontier = list(self.__model__.frontier[:].value)
        return np.asarray(frontier)

In other cases, I think you are right.

**Comment by JulianATA** [comment]
@ds2010 , thanks!
Should we considered about z variables?

**Comment by ds2010** [comment]
Yes, we need to consider it. In that case, the formulation you provide is more concise. So, Taking the z variable into consideration, the frontier is:

      frontier = y/(exp(residual+ lamda*Z)) -1

**Comment by ds2010** [comment]
This issue has been solved in pr #20.
